services/spar/src/Spar/API.hs

```diff
@@ -683,6 +683,7 @@ idpUpdateXML ::
 idpUpdateXML zusr raw idpmeta idpid = withDebugLog "idpUpdate" (Just . show . (^. SAML.idpId)) $ do
   (teamid, idp) <- validateIdPUpdate zusr idpmeta idpid
   GalleyAccess.assertSSOEnabled teamid
+  forM_ (idp ^. SAML.idpExtraInfo . wiOldIssuers) IdPConfigStore.deleteIssuer
   IdPRawMetadataStore.store (idp ^. SAML.idpId) raw
```

Comment:
if spar crashes at the right moment, this will cause the old idp to be deleted and the new not to be introduced, which is not fixable over the rest api (let alone the UI).

I don't see a good way around this, though.  It would be really nice to have transactions and postgres and everything. but i would like to have the comment below reflect this.  (could you maybe also fix my typo there while you're at it ("epected")?)

Reply:
I moved the deletion to the end which leaves a less inconsistent state, in the case of a failure. And I added a comment.